src/core/stream/period/period_stream.ts

```diff
@@ -209,6 +209,11 @@ export default function PeriodStream({
             return reloadAfterSwitch(period, clock$, relativePosAfterSwitch);
           }
 
+          const needsBufferFlush$ =
```

Comment:
As discussed needs a TODO there.

Also, don't we risk to perform the flush even when no data has been buffered here (like for example the first time we're entering a Period?).

This would mean a flush at each Period's beginning and another flush when we're preloading a new  PeriodStream.

Maybe we should keep some of the logic that was there before in `src/core/stream/period/get_adaptation_switch_strategy.ts`